phalcon/Dispatcher/AbstractDispatcher.zep

```diff
@@ -154,10 +155,52 @@ abstract class AbstractDispatcher extends AbstractInjectionAware implements Disp
 
     public function callActionMethod(handler, string actionMethod, array! params = [])
     {
-        return call_user_func_array(
-            [handler, actionMethod],
-            params
+        var result, observer, altHandler, altAction, altParams;
+
+        let altHandler = handler;
+        let altAction = actionMethod;
+        let altParams = params;
+
+        if this->eventsManager !== null && this->eventsManager instanceof ManagerInterface {
+            let observer = <Registry> this->getDi()->get(
```

Comment:
Use a `Collection` object here. Pretty much the same functionality. I have not decided yet if we will keep the registry.